### PR TITLE
Added screenshots base_path support.

### PR DESCRIPTION
**Pull request by kbidarkar**
* Also now the dir structure is based upon the date, this helps
  with quickly clearing screenshots data depending upon the dates.
* `{screenshot_dir}/{date}/{ver}/{dir}/`, looking at the current
  screenshots dir in sesame realized the above dir structure would
  be of immense help.

### CODE REVIEW

File: robottelo.properties.sample [hunk 1]
```diff
@@ -10,6 +10,9 @@ server.hostname=
 
 server.ssh.key_private=/home/whoami/.ssh/id_hudson_dsa
 server.ssh.username=root
+# The directory where screenshots will be saved.
+# Note:- Content under /tmp may be deleted after a reboot.
+screenshot_dir=/tmp/screenshots/
```
Comment (elyezer) [inline robottelo.properties.sample]: I would recommend `/tmp/robottelo/screenshots/` just for the sake of clarity.

Comment (kbidarkar) [inline robottelo.properties.sample]: ok, agreed.

File: robottelo/test.py [hunk 2]
```diff
@@ -238,11 +240,21 @@ def take_screenshot(self, testmethodname):
         testmethodname = "_".join(testmethodname)
         # Creates dir_structure depending upon the testmethodname.
         directory = testmethodname
-        if not os.path.exists("/tmp/screenshots/" + directory):
-            os.makedirs("/tmp/screenshots/" + directory)
-        self.browser.save_screenshot('/tmp/screenshots/{dir}/'
-                                     'screenshot-{date}.png'
-                                     .format(dir=directory, date=filename))
+        os_ver = self.server_name.split(".")[0].split("-")[2]
```
Comment (elyezer) [inline robottelo/test.py]: We can not assume this, because we can have a server name like `ibm-something.domain.com` which does not contains the os version on it. I think for the directory name should be formed by just `{base_dir}/{date}/{dir}`

File: robottelo/test.py [hunk 3]
```diff
@@ -238,11 +240,21 @@ def take_screenshot(self, testmethodname):
...
+        if not os.path.exists('{screenshot_dir}/{date}/{ver}/{dir}/'
+                              .format(screenshot_dir=self.screenshot_dir,
+                                      date=date, ver=os_ver, dir=directory)):
+            os.makedirs('{screenshot_dir}/{date}/{ver}/{dir}/'
```
Comment (elyezer) [inline robottelo/test.py]: This format:

```python
'{screenshot_dir}/{date}/{ver}/{dir}/'.format(screenshot_dir=self.screenshot_dir, date=date,ver=os_ver, dir=directory)
```

Could be saved to a variable for later usage as it is being used more than once

File: robottelo/test.py [hunk 4]
```diff
@@ -238,11 +240,21 @@ def take_screenshot(self, testmethodname):
...
+                        .format(screenshot_dir=self.screenshot_dir, date=date,
+                                ver=os_ver, dir=directory))
+        self.browser.save_screenshot('{screenshot_dir}/{date}/{ver}/{dir}/'
+                                     'screenshot-{dnt}.png'
+                                     .format(
+                                         screenshot_dir=self.screenshot_dir,
+                                         date=date, ver=os_ver,
+                                         dir=directory, dnt=filename,
```
Comment (elyezer) [inline robottelo/test.py]: `dnt=datetime.now().strftime('%Y-%m-%d_%H_%M_%S')` in order to drop single usage variable.

**Comment by elyezer** [comment]
ACK, pending comments.

File: robottelo/test.py (hunk 2 above)
Comment (kbidarkar) [inline robottelo/test.py]: ok, that would help, to take screenshot even for non-jenkins runs.

File: robottelo/test.py [hunk 5]
```diff
@@ -238,11 +238,21 @@ def take_screenshot(self, testmethodname):
         testmethodname = "_".join(testmethodname)
         # Creates dir_structure depending upon the testmethodname.
         directory = testmethodname
-        if not os.path.exists("/tmp/screenshots/" + directory):
-            os.makedirs("/tmp/screenshots/" + directory)
-        self.browser.save_screenshot('/tmp/screenshots/{dir}/'
-                                     'screenshot-{date}.png'
-                                     .format(dir=directory, date=filename))
+        dir_structure = '{screenshots_dir}/{date}/{dir}'.format(
```
Comment (Ichimonji10) [inline robottelo/test.py]: Can we use use `os.path.join` to create this path name? This is fragile. For example, notice how `screenshots_dir` is `/tmp/robottelo/screenshots/` in the config file, which will lead to this output:

    /tmp/robottelo/screenshots//{date}/{dir}

File: robottelo/ui/locators.py [hunk 6]
```diff
@@ -1352,7 +1352,7 @@
         By.XPATH, "//td/a[contains(@href, 'repositories')]"),
 
     # Content views
-    "contentviews.new": (By.XPATH, "//a[@ui-sref='content-views.new']"),
+    "contentviews.new": (By.XPATH, "//a[@ui-sref='content-views.new1']"),
```
Comment (Ichimonji10) [inline robottelo/ui/locators.py]: How is this related to the rest of the pull request?

File: robottelo/test.py [hunk 7]
```diff
@@ -238,11 +238,21 @@ def take_screenshot(self, testmethodname):
...
+        if not os.path.exists(dir_structure):
+            os.makedirs(dir_structure)
+        self.browser.save_screenshot('{dir_structure}/screenshot-{dnt}.png'
+                                     .format(
+                                         dir_structure=dir_structure,
+                                         dnt=datetime.now().strftime(
+                                            '%Y-%m-%d_%H_%M_%S'
+                                         ),
+                                     )
+        )
```
Comment (Ichimonji10) [inline robottelo/test.py]: You can probably elimintate the Travis warnings with something like this:

    self.browser.save_screenshot(
        '{dir_structure}/screenshot-{dnt}.png'.format(
            dir_structure=dir_structure,
            dnt=datetime.now().strftime('%Y-%m-%d_%H_%M_%S'),
        )
    )

File: robottelo/test.py (hunk 5 above)
Comment (elyezer) [inline robottelo/test.py]: :+1:

File: robottelo/test.py (hunk 7 above)
Comment (elyezer) [inline robottelo/test.py]: :+1:

File: robottelo/ui/locators.py (hunk 6 above)
Comment (kbidarkar) [inline robottelo/ui/locators.py]: typo, removing it.

File: robottelo/test.py (hunk 5 above)
Comment (kbidarkar) [inline robottelo/test.py]: ok, will update with os.path.join .

Comment (kbidarkar) [inline robottelo/test.py]: Was just checking, found unix just handles `//` safely. But agreed os.path,join is the best way to do.

File: robottelo/ui/locators.py (hunk 6 above)
Comment (Ichimonji10) [inline robottelo/ui/locators.py]: Thank you.

**Comment by kbidarkar** [comment]
Ready for merge, Tested and works fine. :)

**Comment by Ichimonji10** [comment]
ACK

@elyezer, you want to take a second look?
